src/nvim/path.c

```diff
@@ -481,6 +481,28 @@ static size_t path_expand(garray_T *gap, const char_u *path, int flags)
   return do_path_expand(gap, path, 0, flags, false);
 }
 
+static const char *scandir_next_with_dots(Directory *dir)
+{
+  static int count = 0;
+  const char *entry = NULL;
+  if (dir == NULL) {
+    count = 0;
+  } else {
+    count += 1;
+    if (count == 1) {
+      entry = ".";
+    } else if (count == 2) {
+      entry = "..";
+    } else {
+      entry = os_scandir_next(dir);
+      if (entry == NULL) {
+        count = 0;
```

Comment:
I don't think this is necessary, since `scandir_next_with_dots` should always be initialized before use.

I made some changes in https://github.com/neovim/neovim/commit/efa059c5a4611629890e28987a83aa781952b78c , can you check?